strongbox-web-core/src/main/java/org/carlspring/strongbox/services/DirectoryListingServiceImpl.java

```diff
@@ -131,82 +163,86 @@ private DirectoryListing fromPath(Path path)
+                                 .toUri()
+                                 .toURL();
 
-            files.add(file);
+                file.setUrl(url);
+            }
         }
 
-        Map<String, List<FileContent>> listing = new HashMap<>();
-        listing.put("directories", directories);
-        listing.put("files", files);
-
         return listing;
     }
 
-    /**
-     * @param rootPath
-     *            The root path in which directory listing is allowed. Used as a
-     *            precaution to prevent directory traversing.
-     *            When "path" is outside "rootPath" an exception will be thrown.
-     * @param path
-     *            The path which needs to be listed
-     * @return DirectoryListing
-     * @throws RuntimeException
-     *             when path is not within rootPath.
-     */
-    public DirectoryListing fromPath(Path rootPath,
-                                     Path path)
-        throws IOException
+    private List<Path> fetchContent(Path path)
+            throws IOException
     {
-        rootPath = rootPath.normalize();
-        path = path.normalize();
-
-        if (!path.equals(rootPath) && !path.startsWith(rootPath))
+        List<Path> contentPaths;
+        try (Stream<Path> pathStream = Files.list(path))
         {
-            String message = String.format(
-                                           "Requested directory listing for [%s] is outside the scope of the root path [%s]! Possible intrusion attack or misconfiguration!",
-                                           path, rootPath);
-            logger.error(message);
-            throw new RuntimeException(message);
+            contentPaths = pathStream
+                                   .filter(p -> !p.toFile().getName().startsWith("."))
+                                   .filter(p -> {
+                                       try
+                                       {
+                                           return !Files.isHidden(p);
+                                       }
+                                       catch (IOException e)
+                                       {
+                                           logger.debug("Error accessing path {}", p);
+                                           return false;
+                                       }
+                                   })
+                                   .sorted()
+                                   .collect(Collectors.toList());
         }
 
-        return fromPath(path);
+        return contentPaths;
     }
 
-    private URL calculateDirectoryUrl(FileContent file)
-        throws MalformedURLException
+    private String sanitizedPath(final String rawPath)
     {
-        if (file.getRepositoryId() == null)
-        {
-
-            return new URL(String.format("%s/%s", baseUrl, file.getStorageId()));
+        String sanitizedPath = rawPath;
 
+        if(!File.pathSeparator.equals("/")) {
```

Comment:
Cosmetic: Braces on new lines please. :)
```suggestion
        if(!File.pathSeparator.equals("/"))
        {
```